FinSearchUnified/Bundles/SearchBundleDBAL/ConditionHandler/HasProductNameConditionHandler.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace FinSearchUnified\Bundles\SearchBundleDBAL\ConditionHandler;
+
+use FinSearchUnified\Bundles\SearchBundle\Condition\HasProductNameCondition;
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+use Shopware\Bundle\SearchBundleDBAL\ConditionHandlerInterface;
+use Shopware\Bundle\SearchBundleDBAL\QueryBuilder;
+use Shopware\Bundle\StoreFrontBundle\Struct\ShopContextInterface;
+
+class HasProductNameConditionHandler implements ConditionHandlerInterface
+{
+    /**
+     * @param ConditionInterface $condition
+     *
+     * @return bool
+     */
+    public function supportsCondition(ConditionInterface $condition)
+    {
+        return ($condition instanceof HasProductNameCondition);
+    }
+
+    /**
+     * @param ConditionInterface $condition
+     * @param QueryBuilder $query
+     * @param ShopContextInterface $context
+     *
+     * @return void
```

Comment:
If there was no specific reason to add the `@return`: We always omit it if the method doesn't have a return value.